### ISSUE TITLE
Allowing overwriting number types and its impact.

### ISSUE THREAD

**Issue by sebastiankb** [w3c/wot-thing-description#1054]
Lets assume we have following TM definition: 
```js
{
    "@context": ["http://www.w3.org/ns/td"], 
    "@type" : "ThingModel",
    "title": "Basic On/Off Thing Model",
    "properties": {
        "onOff": {
            "type": "number",
            "minimum" : 0.0,
            "maximum" : 10.5
        }
    }
}
```

and it is extended it will overwrite the onOff:

```js
{
    "@context": ["http://www.w3.org/ns/td"], 
    "@type" : "ThingModel",
    "title": "Smart Lamp Control",
    "links" : [{
        "rel": "extends",
        "href": "http://example.com/BasicOnOffTM",
        "type": "application/td+json"
     }],
    "properties": {
        "onOff": {
            "type": "integer"
        }
    }
}
```

what happen with minimum and maximum?

**Comment by egekorkan** [comment]
I thought that redefining the same property was not allowed with extend? If it is allowed and overwritten with extension, I guess that there should be no relation to the previous case. In this case, the minimum and maximum should not be taken over to the Smart Lamp Control TM.

**Comment by sebastiankb** [comment]
Principle yes, however, number - integer was the once exception. The motivation was provided [here](https://github.com/w3c/wot-thing-description/pull/1024#issuecomment-773891482).

**Comment by relu91** [comment]
If we are going to support the overwrite of types we should enforce somehow the Substitution principle of OOP. This means that extenders could not enforce less strict constraints. Therefore following this rule you can't really overwrite `float` with `integer` because users of the first TM would expect that arguments like `2.2` would be accepted with no modifications. The other way around although it is possible (i.e., `integer` to `float`). The same applies for maximum and minimum you can't enforce a shorter range of values in the overwrite TM.  So the correct way to model this problem and cover the requirements that @sebastiankb described could be done simply like this:
```js
{
    "@context": ["http://www.w3.org/ns/td"], 
    "@type" : "ThingModel",
    "title": "Basic On/Off Thing Model",
  
    "properties": {
        "onOff": {
            "type": "integer",
            "minimum" : 0,
            "maximum" : 10 
        }
    }
}
```
```js
{
    "@context": ["http://www.w3.org/ns/td"], 
    "@type" : "ThingModel",
   "links" : [{
        "rel": "extends",
        "href": "http://example.com/BasicOnOffTM",
        "type": "application/td+json"
     }],
    "title": "Smart Lamp Control",
    "properties": {
        "onOff": {
            "type": "number",
            "minimum" : 0.0,
            "maximum" : 10.5
        }
    }
}
```

Semantically still the extension of the maximum creates problems, but I kinda consider this case as a bad design choice.

**Comment by egekorkan** [comment]
@relu91 I also agree that this generated TM (the second one) is how it should look like. However, I fail to see what I gain by knowing that minimum and maximum are extended. What would be the difference if the root TM had no min and max and I extend it and then add these?

**Comment by sebastiankb** [comment]
@relu91 you are right. I am also tending to take back this exception at the moment and try to keep everything as simple as possible.
